modules/gateway/peers.go

```diff
@@ -58,6 +68,14 @@ func (s insufficientVersionError) Error() string {
 	return "unacceptable version: " + string(s)
 }
 
+// invalidVersionError indicates a peer's version is not a valid version number.
+type invalidVersionError string
+
+// Error implements the error interface for invalidVersionError.
+func (s invalidVersionError) Error() string {
+	return "invalid version: " + string(s)
+}
```

Comment:
The `insufficientVersionError` caught a few errors in testing in the past. I'd prefer to keep this type so as to make testing easier and catch similar errors in the future.